hummingbot/core/event/events.py

```diff
@@ -175,11 +174,21 @@ def order_filled_events_from_order_book_rows(cls,
                                                  order_type: OrderType,
                                                  trade_fee: TradeFeeBase,
                                                  order_book_rows: List[OrderBookRow],
-                                                 exchange_trade_id: str = "") -> List["OrderFilledEvent"]:
+                                                 exchange_trade_id: Optional[str] = None) -> List["OrderFilledEvent"]:
+        if exchange_trade_id is None:
+            exchange_trade_id = order_id
```

Comment:
In this case, we use the `order_id` as base and add a suffix to it to create a trade id, but in the exchanges, we generate the trade id simply as the current timestamp. Why not take the same approach in both, where the generated id is `{order_id}_{int(self._time() * 1e6)}`?

Reply:
Hello Petio. That was my first implementation. But I changed it because since this works inside a loop, and in the context of the paper trade exchange, the execution could be really fast and end up having repeated timestamps (like we have seen in the past for the nonce generator, where we had to add extra logic to advance the number if the timestamp is repeated). If we just have a numerator and we advance it for each trade, then we are sure they will never collide.